### ISSUE TITLE
TextInput Component does not announce "disabled"

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30938]
## Description
When using a screen reader this component does not announce that it is disabled.

This issue is covered in-depth by parent issue #30840, please check there for more details.

## React Native version:
0.63

**Comment by kacieb** [comment]
Issue:
When `accessibilityState={{disabled: true}}`, the TextInput is not disabled. The TextInput ignores this in favor of using the `editable` prop.

This is fine in my opinion - the TextInput is correctly disabled when `editable={true}` on both Android and iOS and is unable to be edited. `accessibilityState` seems to be completely ignored. Perhaps it would be reasonable to add a warning if someone passes in `disabled` that it will be ignored.

Repro Component:
```
export default function Playground(): React.Node {
  const [editable, setEditable] = React.useState(true);
  const [value, setValue] = React.useState('I am a text input!');

  return (
    <View style={styles.container}>
      <Button
        title={
          'press to disable/enable TextInput - currently editable= ' +
          editable.toString()
        }
        onPress={() => {
          setEditable(!editable);
        }}
      />
      <TextInput
        value={value}
        editable={editable}
        accessibilityState={{disabled: true}}
        onChangeText={val => setValue(val)}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    padding: 10,
    paddingTop: 30,
  },
});
```

**Comment by kacieb** [comment]
I think we can close this - should be the same issue as https://github.com/facebook/react-native/issues/30876

**Comment by lunaleaps** [comment]
@nadiia can we close this per @kacieb's comment?

**Comment by lunaleaps** [comment]
Closing due to #30876
